tests/test_plugin_loader.py

```
@@ -4463,5 +4594,79 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+        in sys.modules during execution.  After all threads finish, the
+        temporary entries must be cleaned up and no stale module should remain.
+        """
+        import concurrent.futures
+        import importlib.machinery
+
+        module_names = [f"_mmrelay_test_mod_{i}" for i in range(8)]
+        caught_exceptions = []
+
+        # Create a minimal loader class that doesn't reference __file__
+        class MinimalLoader:
+            def exec_module(self, module):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `exec_module`

Add return type annotation: `None`

(ANN202)